### ISSUE TITLE
i3lock doesn't hide the screen

### ISSUE THREAD

**Issue by vincentbernat** [chjj/compton#234]
I am using i3lock to lock my screen (with the `-i` flag to use an image). Since some months, with Intel drivers, with either X Render backend or glx backend, i3lock doesn't really display the image. The screen is just freezed. If I press a key, the image is displayed. This happens only when compton is running.

**Comment by kdiv** [comment]
I'm also having this issue, I'm using `xss-lock` to trigger `i3lock` (though it happens with every screen locker I have tried, including `slock`), and whenever I wake up the screen with the mouse the screen has the same contents as before sleeping. 

Once the screen is awake, pressing any key makes i3lock paint over the screen as normal. It seems like anything drawn to the screen while the screensaver extension is activated is ignored.

I can reproduce this every time using `sleep 1; xset s activate`.
I'm using `xf86-video-ati 1:7.5.0-2` and compton on default settings. `--xrender-sync-fence` doesn't help in my case.

**Comment by sandsmark** [comment]
fwiw I can reproduce it here as well (and «paint-on-overlay = true;» fixes it).

1. launch «xss-lock -- i3lock»
2. trigger it with «xset s activate».
3. Watch how nothing gets show from it (to hide the contents of the display) until it does its first repaint.

similar results with sxlock, but there only the two bars where I assume it is painting the username and password feedback text are shown.

**Comment by tad-lispy** [comment]
I've run into same issue after switching to GLX backend. Setting `paint-on-overlay = false` in my `~/.config/compton.conf` fixed the problem. Setting `--xrender-sync --xrender-sync-fence` options did not have any effect.

Thank you for sharing your research.

**Comment by MForster** [comment]
Most curiously for me the problem only exists with `paint-on-overlay = false` and goes away with `paint-on-overlay = true`...

**Comment by rpellerin** [comment]
Sorry to re-open this. I'm running into the same issue. I'm triggering i3-lock with `xss-lock` and `xset s 120 120`. The problem does not occur after inactivity timeout but instead when I close the lid. When reopening it, I can still see the content that was previously shown (web browser, etc...) with a transparent i3-lock on top of it. I must type in my password to get back into my session at this point.
None of the above-mentioned solutions worked for me (`--xrender-sync --xrender-sync-fence` or `paint-on-overlay` set to either `false` or `true`).
Thanks

**Comment by helloworlddontecho** [comment]
Here is the solution that worked for me. Using a bash script.
1. Ended the process called xss-lock: 
The way to do it through terminal: ` ps -ef | grep xss-lock | grep -v grep | awk '{print $2}' | xargs kill -9`
You can also dnd the process by using system monitor, htop, etc.
2. Created a bash script.
In my case, I created a text file with the following filename and contents:
Filename:  lockscreen.sh
Content:

```
#!/bin/bash
 
ps -ef | grep compton | grep -v grep | awk '{print $2}' | xargs kill -9 &&  i3lock -n -i  /your/image && nohup compton  &> /dev/null &
```

3.Saved the script/file named lockscreen.sh wherever you like.
4.Make it executable, through terminal (`chmod +x /path/to/lockscreen.sh`)
5. Explaining the script
`ps -ef | grep compton | grep -v grep | awk '{print $2}' | xargs kill -9`   #Kills the process named compton. 
`&&`  Waits until the previous command gets finished before starting the next one
` i3lock -n -i  /your/lockscreen/image `#Starts i3lock. You could also write:       ` systemctl suspend ; i3lock -n -i `     to make it suspend before it starts i3lock. 
` && `Waits until the previous command gets finished before starting the next one
` nohup compton  &> /dev/null &  `   It opens compton again after I unlocked i3lock. It also made me able to run the command many times (if I only wrote `compton` at the end, the computer would only time-out into i3lock once and then it wouldn't be able to do it again. To explain it badly; by writing `nohup compton  &> /dev/null &` ; it runs compton independent from the shell script. 

It is all written in a single line; using &&; to make sure every process finished before the other begins. Doing it in multiple lines always created errors in my case.

After this; i went to the i3 config file
Write in
`exec --no-startup-id xss-lock /path/to/your/lockscreen.sh`
This makes lockscreen.sh the default lockscreen.
I have a timer that locks the computer after 5 minutes of inactivity. This explanation doesnt tell you how to setup a timer like that; I assume you already have. 

You can also use a key binding, I used mod+Shift+s. In my config file, I wrote:

`bindsym $mod+Shift+s exec --no-startup-id /path/to/your/lockscreen.sh`

Restarted i3

Now, everytime the timer locks my computer or I press the right key bingings the computer locks the screen. I can also see a new process called lockscreen.sh that uses 250 KB of RAM. I can end the process, and compton still works properly and I can still lock the screen as described before. However, since it only uses 250 Kb i don't feel the need to end the process.
I can also change the flags in compton ( `compton -i 0.78 --active-opacity 0.95`) and i3; and the script works perfectly well without showing anything in my computer. You would have to change the flags both in the config file if you have compton there and in the lockscreen.sh script. I could also change the final part of the script to suspend 15 s before restarting compton. (`&&  i3lock -n -i  path/to/your/image && sleep 15s && nohup compton  &> /dev/null &`) to be absolutely sure i3lock starts before compton begins, but I didn't need to. You can write && sleep 1s && between every part of the script to be absolutely sure every step is followed through; but again, i didn't need to. 
I copied and pasted parts of the script from elsewhere, don't really know what I'm doing but it worked for me.
Don't forget to change /path/to/your/image and /path/to/your/lockscreen.sh to the actual paths of your lockscreen and images

**Comment by sandsmark** [comment]
I had a couple of fixes in i3lock (402254b5752fcf6a8197b92f558cb0c35289fa74 and 80d4452ec680bcb0e57418f69d44d88ded82047c) that solved it.

it was unfortunately reverted because of issues with some drivers, but you could test pulling in my changes again to see if that solves it for you.

**Comment by ghost** [comment]
@helloworlddontecho i was just passing by, and saw your script, man just `man pkill`
